src/App.js

```diff
@@ -31,53 +37,79 @@ export default function App() {
         <ProductList
           headline={'Burger'}
           product={burgers}
-          category={'burger'}
         />
       </Route>
       <Route path={'/burger/:id'}>
-        <Detailpage category={'burger'} />
+        <Detailpage category={'burger'} onFavoriteClick={toggleFavorite} favorites={favorites}/>
       </Route>
       <Route exact path="/cold-cuts">
         <ProductList
           headline={'Cold-Cuts'}
           product={burgers}
-          category={'burger'}
         />
       </Route>
       <Route exact path="/kebab">
-        <ProductList headline={'Kebab'} product={kebabs} category={'kebab'} />
+        <ProductList headline={'Kebab'} product={kebabs} />
       </Route>
       <Route path={'/kebab/:id'}>
-        <Detailpage category={'kebab'} />
+        <Detailpage category={'kebab'} onFavoriteClick={toggleFavorite}/>
       </Route>
       <Route exact path="/minced">
-        <ProductList headline={'Minced'} product={minced} category={'minced'} />
+        <ProductList headline={'Minced'} product={minced} />
       </Route>
       <Route path={'/minced/:id'}>
-        <Detailpage category={'minced'} />
+        <Detailpage category={'minced'} onFavoriteClick={toggleFavorite}/>
       </Route>
       <Route exact path="/nuggets">
         <ProductList
           headline={'Nuggets'}
           product={nuggets}
-          category={'nuggets'}
         />
       </Route>
       <Route path={'/nuggets/:id'}>
-        <Detailpage category={'nuggets'} />
+        <Detailpage category={'nuggets'} onFavoriteClick={toggleFavorite}/>
       </Route>
       <Route exact path="/sausages">
         <ProductList
           headline={'Sausages'}
           product={sausages}
-          category={'sausages'}
         />
       </Route>
       <Route path={'/sausages/:id'}>
-        <Detailpage category={'sausages'} />
+        <Detailpage category={'sausages'} onFavoriteClick={toggleFavorite}/>
+      </Route>
+      <Route path="/favorites">
+        <ProductList
+          headline={'Favorites'}
+          product={favorites}
+        />
       </Route>
     </Switch>
   )
+  
+  function toggleFavorite(favorite) {
+    const index = favorites.findIndex(favoriteItem => favoriteItem.id === favorite.id)
+    index > -1 ? 
+    setFavorites([
+      ...favorites.slice(0, index),
+      ...favorites.slice(index + 1),
+    ]) : 
+    setFavorites([...favorites, { ...favorite }])
```

Comment:
If favorite is an object of a product already, the curly braces aren't necessary here
```suggestion
    setFavorites([...favorites, ...favorite ])
```